### DISCUSSION TITLE
How to do sql join/relational queries in prisma?

### DISCUSSION THREAD

**Discussion by Vaasu-Dhand** [prisma/prisma#12484] [Q&A]
Hi, I have a fairly complex mssql query. What would be the prisma query for this:
```sql
select x.* from
   (
         select row_number() over (order by JobApplicationId) RowNum, b.ClientId, a.*
         from JobApplications a
         inner join Jobs b
         on a.JobId = b.JobId
         where b.ClientId = ${clientId}
         ${jobId ? `AND a.JobId = ${jobId}` : ''}
         and ((a.ApplicationDate is null)
         or (a.ApplicationDate between '${formatDate(
         from
         )}' and '${formatDate(to)}'))
   ) x
 where RowNum between 1 and 100000
```
Schema:

**Jobs Modal**
```prisma
model Jobs {
  JobId           Int       @id(map: "PK__Jobs__056690C2A0BC1FC0") @default(autoincrement())
  ClientId        Int?
  JobTitle        String?   @db.NVarChar(256)
  LocationCountry String?   @db.NVarChar(256)
  LocationState   String?   @db.NVarChar(256)
  LocationCity    String?   @db.NVarChar(256)
  JobDescription  String?   @db.NVarChar(Max)
  Experience      String?   @db.NVarChar(256)
  Team            String?   @db.NVarChar(256)
  JobType         String?   @db.NVarChar(50)
  SalaryMin       Float?
  SalaryMax       Float?
  PostedDate      DateTime? @db.DateTime
  IsActive        Boolean?
  IsDeleted       Boolean?
  EndDate         DateTime? @db.DateTime
}
```

**JobApplications Modal**

```prisma
model JobApplications {
  JobId                  Int?
  FirstName              String?   @db.NVarChar(256)
  LastName               String?   @db.NVarChar(256)
  EMail                  String?   @db.NVarChar(256)
  Phone                  String?   @db.NVarChar(256)
  CurrentCompany         String?   @db.NVarChar(256)
  LinkedInURL            String?   @db.VarChar(256)
  AddressLine1           String?   @db.VarChar(256)
  AddressLine2           String?   @db.VarChar(256)
  City                   String?   @db.VarChar(256)
  StateProvince          String?   @db.VarChar(256)
  PostalCode             String?   @db.VarChar(256)
  WorkStatus             String?   @db.NVarChar(256)
  ResumeFileName         String?   @db.NVarChar(Max)
  CoverLetterFileName    String?   @db.NVarChar(Max)
  JobApplicationId       Int       @id(map: "PK__JobAppli__BD557F851B73E2C7") @default(autoincrement())
  ApplicationDate        DateTime? @db.DateTime
  JobApplicationRatingId Int?      @default(0, map: "DF__JobApplic__JobAp__75A278F5")
  RatingComments         String?   @db.NVarChar(256)
}
```

**Comment by austincrim** [comment]
Hey @Vaasu-Dhand 👋 ,

Could you help me understand more what the goal of the query is?

FYI, you can always use the [`$queryRaw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryRaw) method to use plain SQL with Prisma.

**Reply by anthonyalayo** [reply]
I am just chiming in as someone who is also using relational queries in Prisma. If we use `$queryRaw`, we lose the benefits of using Prisma in the first place. As far as I know, Prisma's API is designed for NoSQL and doesn't work like JOOQ for SQL databases. Will Prisma ever have a more SQL like builder mode? I made an argument for it [here](https://github.com/prisma/prisma/issues/13017).

**Reply by austincrim** [reply]
Thanks for opening that feature request! It looks like we have a [similar one here](https://github.com/prisma/prisma/issues/5052) as well. Feel free to read through the discussion there and offer any more feedback.

Any more updates will happen in those threads.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
